lib/hd/hd.js

```diff
@@ -46,7 +46,7 @@ HD.generate = function generate() {
 
 /**
  * Generate an {@link HDPrivateKey} from a seed.
- * @param {Object|Mnemonic|Buffer} options - seed,
+ * @param {Buffer} options - seed,
```

Comment:
mine as well remove the word "options" here and rename the actual variable in the code below as well.
Also you can remove the next line ` * mnemonic, mnemonic options.` is leftover junk.

Reply:
The whole file's code follows the same structure with options. Wouldn't it be better if we follow that?

Reply:
`options' almost always means an object in bcoin context, ie.:
```
{
  value: 1,
  thing: "yes",
  configuration: true
}
```

Apparently at some point this function accepted such an options object, and now it does not. Your PR is fixing someone's past mistake here (not updating the comment) but in my opinion the code itself also needs to be fixed.